addons/hw_drivers/controllers/driver.py

```diff
@@ -1,432 +1,382 @@
-        btdrivers.append(newclass)
+        newclass = super(DriverMetaClass, cls).__new__(cls, clsname, bases, attrs)
+        drivers.append(newclass)
         return newclass
 
+class Driver(Thread, metaclass=DriverMetaClass):
+    """
+    Hook to register the driver into the drivers list
+    """
+    connection_type = ""
 
-class BtDriver(Driver, metaclass=BtMetaClass):
-
-
-    def __init__(self, device, manager):
-        super(BtDriver, self).__init__()
+    def __init__(self, device):
+        super(Driver, self).__init__()
         self.dev = device
-        self.manager = manager
-        self.value = ''
+        self.value = ""
+        self.data = {}
         self.gatt_device = False
 
-    def disconnect(self):
-        path = "bt_%s" % (self.dev.mac_address,)
-        del drivers[path]
-
-    def get_name(self):
-        return self.dev.alias()
+    @property
+    def device_name(self):
+        return self._device_name
+
+    @property
+    def device_identifier(self):
+        return self._device_identifier
+
+    @property
+    def device_connection(self):
+        """
+        On specific driver override this method to give connection type of device
+        return string
+        possible value : direct - network - bluetooth
+        """
+        return self._device_connection
+
+    @property
+    def device_type(self):
+        """
+        On specific driver override this method to give type of device
+        return string
+        possible value : printer - camera - device
+        """
+        return self._device_type
+
+    @classmethod
+    def supported(cls, device):
+        """
+        On specific driver override this method to check if device is supported or not
+        return True or False
+        """
+        pass
 
-    def value(self):
-        return self.value
+    def get_message(self):
+        return ''
 
     def action(self, action):
         pass
 
-    def get_connection(self):
-        return 'bluetooth'
+    def disconnect(self):
+        del iot_devices[self.device_identifier]
+
+
+#----------------------------------------------------------
+# Device manager
+#----------------------------------------------------------
+
+class EventManager(object):
+    def __init__(self):
+        self.sessions = {}
+
+    def _delete_expired_sessions(self, max_time=70):
+        '''
+        Clears sessions that are no longer called.
+
+        :param max_time: time a session can stay unused before being deleted
+        '''
+        now = time.time()
+        expired_sessions = [session for session in self.sessions if now - self.sessions[session]['time_request'] > max_time]
+        for session in expired_sessions:
+            del self.sessions[session]
+
+    def add_request(self, listener):
+        self.session = {
+            'session_id': listener['session_id'],
+            'devices': [],
+            'event': Event(),
+            'queue': {},
```

Comment:
i think queue is wronly named here, it is actually the result
you should find a more appropriate name